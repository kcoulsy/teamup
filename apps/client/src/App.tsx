import React from 'react';

import AppRouter from './components/Router/AppRouter';
import 'antd/dist/antd.css';
import './styles.scss';

function App() {
  return <AppRouter />;
}

export default App;
