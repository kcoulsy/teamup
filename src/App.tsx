import React from 'react';
import { connect } from 'react-redux';

import { initialise } from './actions/initialise';

import AppRouter from './components/Router/AppRouter';
import 'antd/dist/antd.css';
import './styles.scss';

interface AppProps {
    initialise: Function;
}

export const App: React.FunctionComponent<AppProps> = ({ initialise }) => {
    initialise();

    return <AppRouter />;
};

const mapDispatchToProps = { initialise };

export default connect(undefined, mapDispatchToProps)(App);
