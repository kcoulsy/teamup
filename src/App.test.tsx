import React from 'react';
import { shallow } from 'enzyme';
import { describe, expect, test, vi } from 'vitest';

import { shallowToJson } from 'enzyme-to-json';
import { App } from './App';

describe('render AppRouter component', () => {
  test('should render AppRouter component', () => {
    const initialise = vi.fn();
    const wrapper = shallow(<App initialise={initialise} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(initialise.mock.calls.length).toBeGreaterThan(0);
  });
});
