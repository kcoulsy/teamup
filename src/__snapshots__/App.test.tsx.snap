// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AppRouter component should render AppRouter component 1`] = `<AppRouter />`;
